### PR TITLE
i3: add notification option to startup submodule

### PR DESCRIPTION
**Pull request by uvNikita**
Fixes https://github.com/rycee/home-manager/issues/129.

### CODE REVIEW

File: modules/services/window-managers/i3.nix [hunk 1]
```diff
@@ -20,6 +20,15 @@ let
         description = "Whether to run command on each i3 restart.";
       };
 
+      notification = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether to enable startup-notification support for the command.
+          See --no-startup-id option description in the i3 user guide.
```
Comment (rycee) [inline modules/services/window-managers/i3.nix]: Can use the [option](http://tdg.docbook.org/tdg/5.2/option.html) docbook element: `<option>--no-startup-id</option>`.

File: modules/misc/news.nix [hunk 2]
```diff
@@ -448,6 +448,20 @@ in
...
         '';
       }
+      {
+        time = "2017-11-14T19:56:49+00:00";
+        condition = with config.xsession.windowManager; (
+          i3.enable && i3.config != null && i3.config.startup != []
+        );
+        message = ''
+          A new 'notification' option was added to
+          config.xsession.windowManager.i3.startup submodule.
```
Comment (rycee) [inline modules/misc/news.nix]: I think you can remove the `config.` in `config.xsession.windowManager.i3.startup`.

**Comment by rycee** [comment]
Looks good to me. Just two minor comments.

File: modules/services/window-managers/i3.nix (hunk 1 above)
Comment (uvNikita) [reply modules/services/window-managers/i3.nix]: This works well when I add it in the parameter description, but in the news, it just renders like `<option>--no-startup-id</option>`. Am I doing something wrong?

Comment (rycee) [reply modules/services/window-managers/i3.nix]: Yeah, you'll have to keep it as `--no-startup-id` in the news entry. The news system is still very rudimentary and will not attempt to do any formatting of the message text. I.e., what you write there is shown as-is.

I'd like to make it support formatting at some point but haven't felt that it's very urgent 😉

Comment (t184256) [reply modules/services/window-managers/i3.nix]: Maybe you could strip the formatting in the meantime.

Comment (rycee) [reply modules/services/window-managers/i3.nix]: I think it'll be a bit tricky to reliably remove the formatting and still keep pretty plain text output. For now I think it's easier to just write plain text in the news entries and then when some markup is integrated we can just convert the news entries to the chosen markup.

**Comment by uvNikita** [comment]
Made suggested changes and merged into master: https://github.com/rycee/home-manager/commit/1946343d5b1990539c383c2edbf529deed2e6916. Thanks for the review!

**Pull request closed without merging**